### PR TITLE
fix(@angular-devkit/build-angular): Sass compilation in StackBlitz webcontainers

### PR DESCRIPTION
**Pull request by alan-agius4**
When `process.versions.webcontainer` is a string it means that we are running in a StackBlitz webcontainer. `SassWorkerImplementation` uses `receiveMessageOnPort` Node.js `worker_thread` API to ensure sync behavior which is ~2x faster. However, it is non trivial to support this in a webcontainer and while slower we choose to use `dart-sass` which in Webpack uses the slower async path.

// cc @markwhitfeld

### CODE REVIEW

**Review by markwhitfeld — APPROVED** [review]
That check looks good 👍

**Comment by d3lm** [comment]
Thanks so much for this quick implementation and for adding an env check here for the time being. I am actually gonna look into this very soon so we can support `receiveMessageOnPort` at least minimally. Thanks a lot!

File: packages/angular_devkit/build_angular/src/webpack/configs/styles.ts [hunk 1]
```diff
@@ -404,3 +406,14 @@ export function getStylesConfig(wco: WebpackConfigOptions): Configuration {
...
+
+function getSassImplementation(): SassWorkerImplementation | typeof import('sass') {
+  const { webcontainer } = process.versions as unknown as Record<string, unknown>;
+
+  // When `webcontainer` is a string it means that we are running in a StackBlitz webcontainer.
+  // `SassWorkerImplementation` uses `receiveMessageOnPort` Node.js `worker_thread` API to ensure sync behavior which is ~2x faster.
+  // However, it is non trivial to support this in a webcontainer and while slower we choose to use `dart-sass`
+  // which in Webpack uses the slower async path.
+  // We should periodically check with StackBlitz folks (Mark Whitfeld / Dominic Elm) to determine if this workaround is still needed.
+  return typeof webcontainer === 'string' ? require('sass') : new SassWorkerImplementation();
```
Comment (dgp1130) [inline packages/angular_devkit/build_angular/src/webpack/configs/styles.ts:418]: Nit: Do we care that `webcontainer` is a `string`? Would it be safer/cleaner to just check if the property exists / is truthy?

**Review by dgp1130 — APPROVED** [review]
Just for context, do we only use `dart-sass` for webcontainers, or are there other usages were it is still supported? I'm just wondering if this is the major thing blocking us from moving to `node-sass`, or if there are other problems that need to be resolved anyways.

File: packages/angular_devkit/build_angular/src/webpack/configs/styles.ts (hunk 1 above)
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/webpack/configs/styles.ts:418]: I think we can check if it’s truthy. But  stackblitz folks suggested this way.

@markwhitfeld thoughts?

**Comment by alan-agius4** [comment]
> Just for context, do we only use `dart-sass` for webcontainers, or are there other usages were it is still supported? I'm just wondering if this is the major thing blocking us from moving to `node-sass`, or if there are other problems that need to be resolved anyways.

We only support dart-sass.

**Comment by dgp1130** [comment]
> We only support dart-sass.

Ah thanks, I was getting the direction of `node-sass` -> `dart-sass` backwards. Disabling the worker in webcontainers is reasonable until it has the necessary support.

File: packages/angular_devkit/build_angular/src/webpack/configs/styles.ts (hunk 1 above)
Comment (d3lm) [reply packages/angular_devkit/build_angular/src/webpack/configs/styles.ts:418]: The value will be a commit sha and it should be a string yes. I think either works.

File: packages/angular_devkit/build_angular/src/webpack/configs/styles.ts [hunk 2]
```diff
@@ -404,3 +406,14 @@ export function getStylesConfig(wco: WebpackConfigOptions): Configuration {
...
+
+function getSassImplementation(): SassWorkerImplementation | typeof import('sass') {
+  const { webcontainer } = process.versions as unknown as Record<string, unknown>;
+
+  // When `webcontainer` is a truthy it means that we are running in a StackBlitz webcontainer.
+  // `SassWorkerImplementation` uses `receiveMessageOnPort` Node.js `worker_thread` API to ensure sync behavior which is ~2x faster.
+  // However, it is non trivial to support this in a webcontainer and while slower we choose to use `dart-sass`
+  // which in Webpack uses the slower async path.
+  // We should periodically check with StackBlitz folks (Mark Whitfeld / Dominic Elm) to determine if this workaround is still needed.
+  return webcontainer ? require('sass') : new SassWorkerImplementation();
```
Comment (d3lm) [inline packages/angular_devkit/build_angular/src/webpack/configs/styles.ts:418]: Maybe we can make this more explicit and say `webcontainer != null`, WDYT?

**Pull request merged**